Objects/obmalloc.c

```diff
@@ -604,13 +612,17 @@ _PyMem_GetCurrentAllocatorName(void)
 }
 
 
-#ifdef WITH_PYMALLOC
-static int
+int
 _PyMem_DebugEnabled(void)
 {
-    return (_PyObject.malloc == _PyMem_DebugMalloc);
+#ifdef WITH_PYMALLOC
```

Comment:
It seems like this ifdef isn't necessary and i should always reutnr `is_debug_enabled`?  This flag gets set regardless if Python is built w/ pymalloc or without.  It also seems like maybe builds with the GIL disabled will ultimately want to build w/o PYMALLOC given that it's not suitable for them?